src/MgvOfferTaking.sol

```diff
@@ -515,7 +517,9 @@ abstract contract MgvOfferTaking is MgvHasOffers {
         (mgvData, gasused, makerData) = innerDecode(retdata);
         /* Note that in the `if`s, the literals are bytes32 (stack values), while as revert arguments, they are strings (memory pointers). */
         if (mgvData == "mgv/makerRevert" || mgvData == "mgv/makerTransferFail" || mgvData == "mgv/makerReceiveFail") {
-          emit OfferFail(sor.outbound_tkn, sor.inbound_tkn, sor.offerId, mor.taker, sor.wants, sor.gives, mgvData);
+          // FIXME: Why don't we pay the penalty here? Because we may use more gas and thus incur a bigger penalty, according to a comment higher up.
```

Comment:
I like the idea of emitting `OfferFail` later a lot since it can now include the penalty, but being out-of-order with the posthook events seems really bad...

Reply:
Indeed. I experimented with a few a solutions, but ended up preferring the separate `Penalty` event as it preserves the ordering.